filters/include/pcl/filters/impl/crop_hull.hpp

```diff
@@ -40,37 +40,24 @@
 
 #include <pcl/filters/crop_hull.h>
 
+
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 template<typename PointT> void
 pcl::CropHull<PointT>::applyFilter (PointCloud &output)
 {
-  if (dim_ == 2)
-  {
-    // in this case we are assuming all the points lie in the same plane as the
-    // 2D convex hull, so the choice of projection just changes the
-    // conditioning of the problem: choose to squash the XYZ component of the
-    // hull-points that has least variation - this will also give reasonable
-    // results if the points don't lie exactly in the same plane
-    const Eigen::Vector3f range = getHullCloudRange ();
-    if (range[0] <= range[1] && range[0] <= range[2])
-      applyFilter2D<1,2> (output);
-    else if (range[1] <= range[2] && range[1] <= range[0])
-      applyFilter2D<2,0> (output);
-    else
-      applyFilter2D<0,1> (output);
-  }
-  else
-  {
-    applyFilter3D (output);
-  }
+  //TODO: remove overriden function with trivial reimplementation
```

Comment:
https://godbolt.org/z/vsKcWKMeP

Reply:
Do I understand correctly? Are you want some like this:
```cpp
PCL_DEPRECATED(1, 12, "This method is trivially call Base and must be removed")
```

Reply:
Yeah. Something like: "This is a trivial call to base class method" The removal message is auto-added by the macro

Reply:
done

Reply:
but I have some problem with message warnings:
1. if current version less than deprecated target, warning are not exists
1. else:
```
/home/ros/tmp/pcl/common/include/pcl/pcl_macros.h:162:11: error: 'unneeded_deprecation' does not name a type
           unneeded_deprecation),                                                       \
           ^
/home/ros/tmp/pcl/common/include/pcl/pcl_macros.h:157:3: note: in expansion of macro '_PCL_COMPAT_MAJOR_VERSION'
   _PCL_COMPAT_MAJOR_VERSION(                                                           \
   ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/ros/tmp/pcl/common/include/pcl/pcl_macros.h:159:7: note: in expansion of macro '_PCL_COMPAT_MINOR_VERSION'
       _PCL_COMPAT_MINOR_VERSION(                                                       \
       ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/ros/tmp/pcl/filters/include/pcl/filters/impl/crop_hull.hpp:46:1: note: in expansion of macro 'PCL_DEPRECATED'
 PCL_DEPRECATED(1, 12, "This is a trivial call to base class method")
 ```
 
 I have tested also with already exists macros and behavior are same:
 ```
 /home/ros/tmp/pcl/common/include/pcl/pcl_macros.h:162:11: error: 'unneeded_deprecation' does not name a type
           unneeded_deprecation),                                                       \
           ^
/home/ros/tmp/pcl/common/include/pcl/pcl_macros.h:157:3: note: in expansion of macro '_PCL_COMPAT_MAJOR_VERSION'
   _PCL_COMPAT_MAJOR_VERSION(                                                           \
   ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/ros/tmp/pcl/common/include/pcl/pcl_macros.h:159:7: note: in expansion of macro '_PCL_COMPAT_MINOR_VERSION'
       _PCL_COMPAT_MINOR_VERSION(                                                       \
       ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/ros/tmp/pcl/filters/include/pcl/filters/passthrough.h:154:7: note: in expansion of macro 'PCL_DEPRECATED'
       PCL_DEPRECATED(1, 12, "use inherited FilterIndices::setNegative() instead")
```

Reply:
Yes. The version in the deprecated macro are when this needs to be removed.